url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/model3d/TetrahedronNet3DComp.java

```diff
@@ -0,0 +1,2632 @@
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 */
+	public boolean containsElt(Segment3D seg) { // Dag
+		ScalarOperator sop = this.getScalarOperator();
+		Set set = this.getSAM().intersects(seg.getMBB());
+		set = this.getSegments(set);
+
+		Iterator it = set.iterator();
+		while (it.hasNext()) {
+			Segment3D segment = (Segment3D) it.next();
+			if (segment.isGeometryEquivalent(seg, sop))
+				return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Tests whether a element with the coordinates of given point is contained
+	 * in the component.<br>
+	 * Identity test based on epsilon equality on coordinates !
+	 * 
+	 * @param point
+	 *            Point3D
+	 * @return boolean - true if contained, false otherwise.
+	 * @throws IllegalArgumentException
+	 *             if the index of the point of the tetrahedron is not in the
+	 *             interval [0;3]. The exception originates in the method
+	 *             getPoint(int) of the class Tetrahedron3D.
+	 */
+	public boolean containsElt(Point3D point) { // Dag
+		ScalarOperator sop = this.getScalarOperator();
+		Set set = this.getSAM().intersects(point.getMBB());
+		set = this.getPoints(set);
+
+		Iterator it = set.iterator();
+		while (it.hasNext()) {
+			Point3D p = (Point3D) it.next();
+			if (p.isEqual(point, sop))
+				return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Returns the number of elements in the net.
+	 * 
+	 * @return int - number of elements.
+	 */
+	public int countElements() {
+		return getSAM().getCount();
+	}
+
+	/**
+	 * Returns the number of vertices in this component.
+	 * 
+	 * @return int - number of vertices.
+	 */
+	public int countVertices() {
+		return getVertices();
+	}
+
+	/**
+	 * Returns the number of edges in this component.
+	 * 
+	 * @return int - number of edges.
+	 */
+	public int countEdges() {
+		return getEdges();
+	}
+
+	/**
+	 * Returns the number of faces in this component.
+	 * 
+	 * @return int - number if faces.
+	 */
+	public int countFaces() {
+		return getFaces();
+	}
+
+	/**
+	 * Returns the number of tetrahedrons in this component.
+	 * 
+	 * @return int - number of tetrahedrons.
+	 */
+	public int countTetras() {
+		return getSAM().getCount();
+	}
+
+	/*
+	 * Returns the number of elements of specified type in the border of this
+	 * component. Possible types of elements to count are: tetrahedrons,
+	 * triangles, segments or vertices.
+	 * 
+	 * @param typ SimpleGeoObj constants
+	 * 
+	 * @return int - count of elements of the specified type.
+	 * 
+	 * @throws IllegalArgumentException - if index of a triangle point is not 0,
+	 * 1 or 2. The exception originates in the method getPoint(int) of the class
+	 * Triangle3D.
+	 * 
+	 * @throws ArithmeticException - if norm equals zero in epsilon range. This
+	 * exception originates in the method normalize(ScalarOperator) of the class
+	 * Vector3D.
+	 */
+	private int countBorderElements(byte typ) { // Dag
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 648-665, lines 670-691 into separate methods.